experimental/dds/tree2/docs/undo/v1-undo.md

```diff
@@ -167,29 +162,33 @@ The alternative, coupling repair data with the changesets that birthed them, is
     making it more self-contained and therefore easier to implement and test.
     This is made possible by the fact that repair data is **not** required for the purpose of rebasing changesets.
 
-#### Implementation Strategy
+### The Repair Data `TreeIndex`
 
-Each `StygianForest` wraps a "normal" forest, meaning, a forest that is unaware of the distinction between in-document and deleted nodes.
-From hereon, we refer to this wrapped forest as "the inner forest".
+One challenge associated with storing repair data in the forest,
+is that it requires us to establish a system for keeping track of which removed subtrees exist in the repair data,
+and where they reside in the forest.
+Specifically, we have to maintain a correspondence between the identification scheme used by changesets, and the one used by the forest.
+Changesets identify removed trees using a `ChangeAtomId` while the forest uses paths.
```

Comment:
What's the meaning behind this name?

Reply:
When we have a change that deletes 3 nodes in a row in a sequence field, we represent that with a single Delete mark of length 3, but that's just an optimisation for the fact that there are 3 deletions. We've used the term "change atom" to refer the the smallest unit of change that a changeset can be made of. In that "delete 3 nodes" scenario, we allocate 3 `ChangeAtomId`s for the operation, but because we allocate them as one block, we only have to write down the `ChangeAtomId` for the first deletion and the rest can be infered. Identifying the change in this way also lets us identify the affected nodes using the same ID.